starport/services/networkbuilder/verification.go

```diff
@@ -0,0 +1,118 @@
+package networkbuilder
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/starport/starport/pkg/jsondoc"
+)
+
+type VerificationError struct {
+	Err error
+}
+
+func (err VerificationError) Error() string {
+	return err.Err.Error()
+}
+
+type gentxInfo struct {
+	ValidatorAddress string
+	SelfDelegation   sdk.Coin
+}
+
+// VerifyProposals if proposals are correct and simulate them with the current launch information
+// Correctness means checks that have to be performed off-chain
+func (b *Builder) VerifyProposals(ctx context.Context, chainID string, homeDir string, proposals []int, commandOut io.Writer) error {
+
+	// Check all proposal
+	for _, id := range proposals {
+		proposal, err := b.ProposalGet(ctx, chainID, id)
+		if err != nil {
+			return err
+		}
+
+		// If this is a add validator proposal
+		if proposal.Validator != nil {
```

Comment:
```suggestion
		if proposal.Validator == nil { continue }
```

to avoid another nested block.